pennylane/transforms/zx/converter.py

```diff
@@ -0,0 +1,577 @@
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+class VertexType:  # pylint: disable=too-few-public-methods
+    """Type of a vertex in the graph.
+
+    This class is copied from PyZX as we do not make PyZX a Pennylane requirement.
+
+    Copyright (C) 2018 - Aleks Kissinger and John van de Wetering"""
+
+    BOUNDARY = 0
+    Z = 1
+    X = 2
+    H_BOX = 3
+
+
+class EdgeType:  # pylint: disable=too-few-public-methods
+    """Type of an edge in the graph.
+
+    This class is copied from PyZX as we do not make PyZX a Pennylane requirement.
+
+    Copyright (C) 2018 - Aleks Kissinger and John van de Wetering"""
+
+    SIMPLE = 1
+    HADAMARD = 2
+
+
+@op_transform
+def to_zx(qscript, expand_measurement=False):  # pylint: disable=unused-argument
+    """This transform converts a PennyLane quantum script to a ZX-Graph in the `PyZX framework <https://pyzx.readthedocs.io/en/latest/>`_.
+    The graph can be optimized and transformed by well-known ZX-calculus reductions.
+
+    Args:
+        qscript(QuantumScript, Operation): The PennyLane quantum circuit.
+        expand_measurements(bool): The expansion will be applied on measurements that are not in the Z-basis and
+            rotations will be added to the operations.
+
+    **Example**
+
+    You can use the transform decorator directly on your :class:`~.QNode`, quantum function and executing it will produce a
+    PyZX graph. You can also use the transform directly on the :class:`~.QuantumScript`.
+
+    .. code-block:: python
+
+        import pyzx
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.transforms.to_zx
+        def circuit(p):
+            qml.RZ(p[0], wires=1),
+            qml.RZ(p[1], wires=1),
+            qml.RX(p[2], wires=0),
+            qml.PauliZ(wires=0),
+            qml.RZ(p[3], wires=1),
+            qml.PauliX(wires=1),
+            qml.CNOT(wires=[0, 1]),
+            qml.CNOT(wires=[1, 0]),
+            qml.SWAP(wires=[0, 1]),
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        params = [5 / 4 * np.pi, 3 / 4 * np.pi, 0.1, 0.3]
+        g = circuit(params)
+
+    >>> g
+    Graph(22 vertices, 24 edges)
+
+    It is now a PyZX graph and can apply function from the framework on your Graph, for example you can draw it:
+
+    >>> pyzx.draw(g)
+    <Figure size 800x200 with 1 Axes>
+
+    Alternatively you can use the transform directly on a quantum script and get PyZX graph.
+
+    .. code-block:: python
+
+        operations = [
+                qml.RZ(5 / 4 * np.pi, wires=1),
+                qml.RZ(3 / 4 * np.pi, wires=1),
+                qml.RX(0.1, wires=0),
+                qml.PauliZ(wires=0),
+                qml.RZ(0.3, wires=1),
+                qml.PauliX(wires=1),
+                qml.CNOT(wires=[0, 1]),
+                qml.CNOT(wires=[1, 0]),
+                qml.SWAP(wires=[0, 1]),
+            ]
+
+        qscript = QuantumScript(operations)
```

Comment:
```suggestion
        qscript = qml.tape.QuantumScript(operations)
```

Since readers may not know the path to create this object.  Thanks for this example!